cpu/nrf52/radio/nrf802154/nrf802154_radio.c

```diff
@@ -0,0 +1,721 @@
+            len += iolist->iol_len;
+        }
+    }
+
+    DEBUG("[nrf802154] send: putting %i bytes into the frame buffer\n", len);
+
+    /* specify the length of the package. */
+    txbuf[0] = len + IEEE802154_FCS_LEN;
+
+    return len;
+}
+
+static int _confirm_transmit(ieee802154_dev_t *dev, ieee802154_tx_info_t *info)
+{
+    (void) dev;
+
+    if (_state != STATE_IDLE && _state != STATE_CCA_BUSY && NRF_RADIO->STATE != RADIO_STATE_STATE_Disabled) {
+        return -EAGAIN;
+    }
+
+    if (info) {
+        info->status = (_state == STATE_CCA_BUSY) ? TX_STATUS_MEDIUM_BUSY : TX_STATUS_SUCCESS;
+    }
+
+    _state = STATE_IDLE;
+    NRF_RADIO->SHORTS = DEFAULT_SHORTS;
+
+    return 0;
+}
+
+static int _request_transmit(ieee802154_dev_t *dev)
+{
+    (void) dev;
+    if (cfg.ifs) {
+        return -EBUSY;
+    }
+
+    _state = STATE_TX;
+    if (cfg.cca_send) {
+        NRF_RADIO->SHORTS = CCA_SHORTS;
+        NRF_RADIO->TASKS_RXEN = 1;
+    }
+    else {
+        NRF_RADIO->TASKS_TXEN = 1;
+    }
+
+    return 0;
+}
+
+/**
+ * @brief   Convert from the internal representation to dBm, when the
+ *          radio operates as a IEEE802.15.4 transceiver.
+ */
+static inline int8_t _hwval_to_ieee802154_dbm(uint8_t hwval)
+{
+    return (ED_RSSISCALE * hwval) + ED_RSSIOFFS;
+}
+
+static int _indication_rx(ieee802154_dev_t *dev, void *buf, size_t max_size, ieee802154_rx_info_t *info)
+{
+    (void) dev;
+    size_t pktlen = (size_t)rxbuf[0] - IEEE802154_FCS_LEN;
+    int res = -ENOBUFS;
+
+    if (max_size < pktlen) {
+        DEBUG("[nrf802154] recv: buffer is to small\n");
+        return res;
+    }
+    else {
+        DEBUG("[nrf802154] recv: reading packet of length %i\n", pktlen);
+        if (info != NULL) {
+            ieee802154_rx_info_t *radio_info = info;
+            /* Hardware link quality indicator */
+            uint8_t hwlqi = rxbuf[pktlen + 1];
+            /* Convert to 802.15.4 LQI (page 319 of product spec v1.1) */
+            radio_info->lqi = (uint8_t)(hwlqi > UINT8_MAX/ED_RSSISCALE
+                                       ? UINT8_MAX
+                                       : hwlqi * ED_RSSISCALE);
+            /* Calculate RSSI by subtracting the offset from the datasheet.
+             * Intentionally using a different calculation than the one from
+             * figure 122 of the v1.1 product specification. This appears to
+             * match real world performance better */
+            radio_info->rssi = _hwval_to_ieee802154_dbm(hwlqi) + IEEE802154_RADIO_RSSI_OFFSET;
+        }
+        memcpy(buf, &rxbuf[1], pktlen);
+    }
+
+    NRF_RADIO->TASKS_START = 1;
+
+    return pktlen;
+}
+
+static int _confirm_cca(ieee802154_dev_t *dev)
+{
+    (void) dev;
+    int res;
+    switch (_state) {
+        case STATE_CCA_CLEAR:
+            res = true;
+            break;
+        case STATE_CCA_BUSY:
+            res = false;
+            break;
+        default:
+            res = -EAGAIN;
+    }
```

Comment:
```suggestion
    switch (_state) {
    case STATE_CCA_CLEAR:
        res = true;
        break;
    case STATE_CCA_BUSY:
        res = false;
        break;
    default:
        res = -EAGAIN;
    }
```

as per coding convention ;)